qml/pages/ChatPage.qml

```diff
@@ -660,22 +664,15 @@ Page {
                                         sourceComponent: imagePreviewComponent
                                     }
 
-                                    Component {
-                                        id: stickerPreviewComponent
-                                        StickerPreview {
-                                            id: messageStickerPreview
-                                            stickerData: ( display.content['@type'] === "messageSticker" ) ?  display.content.sticker : ""
-                                            visible: display.content['@type'] === "messageSticker"
-                                            anchors.horizontalCenter: parent.horizontalCenter
-                                        }
-                                    }
-
                                     Loader {
-                                        id: stickerPreviewLoader
-                                        active: false
+                                        x: messageListItem.isOwnMessage ? (parent.width - width) : 0
+                                        active: messageListItem.isSticker
                                         asynchronous: true
-                                        width: parent.width
-                                        sourceComponent: stickerPreviewComponent
+                                        sourceComponent: Component {
```

Comment:
Also here, if you do it, please do it consistently. I, however, prefer the current solution. ;)

Reply:
The "current" meaning the way I did it or the way you did it originally?

Note that I did it in one of the ways [recommended](https://doc.qt.io/qt-5/qml-qtquick-loader.html#using-a-loader-within-a-view-delegate) by Qt. If there's a reason to separate the component from the loader, they recommend moving it out of the delegate, outside of the view (possibly for performance reasons but I'm not sure about that).

Reply:
The way it's there originally, I simply don't like nested components that much. But yes, I'm also aware of the recommendations by Qt. From my perspective there are no issues until you want to access any context-specific properties such as `index`...

Reply:
Let me try to convince you one more time.

1. Think of how loadable components are being positioned. When you define component and loader separately, it takes additional mental efforts to keep in mind that this particular component is being position according to the rules defined for the loader component. IMO it's easier to read and generally cleaner when the component and its positioning rules are right next to each other, inside the same component hierarchy.
2. It's more compact. Note that the way I wrote it, neither loader not the component need to have an id. There are also fewer bindings which is good for performance (although to estimate the impact on performance, one has to keep in mind both number of bindings and complexity of expressions).

If you still want them separate from each other, well, then so be it, I can do it that way.

Reply:
This one is probably also not my battle. If you want to introduce nested components, I'll also merge them probably - if it's done consistently. :)

Reply:
ok, cool. I wouldn't want to do a lot of work just for you to reject it.

Reply:
:+1: